src/SpendPermissionManager.sol

```diff
@@ -237,6 +243,7 @@ contract SpendPermissionManager is EIP712 {
         }
         // loop through each unique spend permission in the batch and approve it
         uint256 batchLen = spendPermissionBatch.permissions.length;
+        if (batchLen == 0) revert EmptyBatch(batchHash);
```

Comment:
curious what benefits you see of putting the check here versus in `getBatchHash`?

Reply:
also agree that `getBatchHash` is the better place